### ISSUE TITLE
version 1.8 drops symbol cram_nop_decode_reset without bumping soversion

### ISSUE THREAD

**Issue by tillea** [samtools/htslib#695]
Hi,
I'm about to build Debian packages of htslib 1.8.  I realised that in file cram/cram_codecs.c function

    void cram_nop_decode_reset(cram_codec *c) {}

was droped.  Well, droping an empty function looks harmless but you never know what programmers out there did with this function.  So its de facto an ABI change which should be reflected in bumping soversion.
Kind regards, Andreas.

**Comment by jkbonfield** [comment]
While technically available in the symbol table, this is an internal function with no public prototype.  Yes you could go grubbing around the library to find symbols you can utilise, but it's not something supported and you'd be on your own if you did so.

Hence we don't regard this as an ABI change as it isn't part of our public ABI.

**Comment by mapreri** [comment]
Yes, GCC and clang follow the same specs (after all, clang aims at being pretty mucha a drop in replacement in most cases...), ISTR msvc has a slightly different syntax, but supporting both is not particularly difficult.

Could you please consider reopening and retitling this bug report?
